hivemq-edge/src/frontend/src/extensions/datahub/components/panels/OperationPanel.tsx

```diff
@@ -1,19 +1,38 @@
-import { FC, useMemo } from 'react'
+import { FC, useCallback, useMemo } from 'react'
 import { Node } from 'reactflow'
 import { Card, CardBody } from '@chakra-ui/react'
+import { IChangeEvent } from '@rjsf/core'
 
+import { MOCK_OPERATION_SCHEMA } from '../../api/specs/'
 import { OperationData, PanelProps } from '../../types.ts'
-import { MOCK_OPERATION_SCHEMA } from '../../api/specs/OperationData.ts'
 import useDataHubDraftStore from '../../hooks/useDataHubDraftStore.ts'
 import { ReactFlowSchemaForm, datahubRJSFWidgets } from '../helpers'
 
 export const OperationPanel: FC<PanelProps> = ({ selectedNode, onFormSubmit }) => {
-  const { nodes } = useDataHubDraftStore()
+  const { nodes, functions } = useDataHubDraftStore()
 
   const formData = useMemo(() => {
     const adapterNode = nodes.find((e) => e.id === selectedNode) as Node<OperationData> | undefined
-    return adapterNode ? adapterNode.data : null
-  }, [selectedNode, nodes])
+    if (!adapterNode?.data) return null
+    return adapterNode?.data
+  }, [nodes, selectedNode])
+
+  const onFixFormSubmit = useCallback(
+    (data: IChangeEvent<OperationData>) => {
+      const initData = data
+      const { formData } = initData
+      if (formData) {
+        const { functionId } = formData
+        const fct = functions.find((e) => e.functionId === functionId)
+        if (fct) {
+          const { metadata } = fct
+          initData.formData = { ...initData.formData, metadata }
+        }
+      }
+      onFormSubmit?.(initData)
```

Comment:
praise: 🎩 another optional function call gentlemen

```suggestion
      onFormSubmit?.(initData)
```